mvrx-compose/src/main/kotlin/com/airbnb/mvrx/compose/MavericksComposeExtensions.kt

```diff
@@ -0,0 +1,113 @@
+package com.airbnb.mvrx.compose
+
+import androidx.activity.ComponentActivity
+import androidx.annotation.RestrictTo
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.remember
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.platform.LocalSavedStateRegistryOwner
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.lifecycle.viewmodel.compose.LocalViewModelStoreOwner
+import androidx.savedstate.SavedStateRegistryOwner
+import com.airbnb.mvrx.ActivityViewModelContext
+import com.airbnb.mvrx.FragmentViewModelContext
+import com.airbnb.mvrx.Mavericks
+import com.airbnb.mvrx.MavericksState
+import com.airbnb.mvrx.MavericksViewModel
+import com.airbnb.mvrx.MavericksViewModelProvider
+import com.airbnb.mvrx.withState
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import kotlin.reflect.KProperty1
+
+/**
+ * Creates a Compose State variable that will emit new values whenever this ViewModel's state changes.
+ * Prefer the overload with a state property reference to ensure that your composable only recomposes when the properties it uses changes.
+ */
+@Composable
+fun <VM : MavericksViewModel<S>, S : MavericksState> VM.collectState(): State<S> {
+    return stateFlow.collectAsState(initial = withState(this) { it })
+}
+
+/**
+ * Creates a Compose State variable that will only update when the value of this property changes.
+ * Prefer this to subscribing to entire state classes which will trigger a recomposition whenever any state variable changes.
+ * If you find yourself subscribing to many state properties in a single composable, consider breaking it up into smaller ones.
+ */
+@Composable
+fun <VM : MavericksViewModel<S>, S : MavericksState, A> VM.collectState(prop1: KProperty1<S, A>): State<A> {
```

Comment:
have you considered a syntax with a lambda with state as receiver so we could do `vm.collectState { myProp }`

I think this would have the benefits of:
- shorter from not invoking the state class name
- maybe faster, with no reflection
- easily allow watching nested properties `vm.collectState { myProp.nestedProp }`

Reply:
Added!